Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -2113,7 +2113,13 @@ extension BrowserViewController: ContextMenuHelperDelegate {
             let copyAction = UIAlertAction(title: copyImageTitle, style: UIAlertActionStyle.Default) { (action: UIAlertAction) -> Void in
                 let pasteBoard = UIPasteboard.generalPasteboard()
                 pasteBoard.string = url.absoluteString
-                // TODO: put the actual image on the clipboard
+                // put the actual image on the clipboard
+                // do this be asyncronously just in case we're in a low bandwidth situation
+                // Is this neccessary, or is the image already sat in cache?
+                NSURLSession.sharedSession().dataTaskWithURL(url, completionHandler: { (data, _, error) -> Void in
```

Comment:
What happens if the user copies an image to their clipboard then backgrounds the app right away? I feel like this would be a common use case and I'm not sure if the callback will get called since the app is in the background. Do we need to wrap this with a Background Task handler?